### DISCUSSION TITLE
[IIC2133] - Cápsula 08: Valgrind - Duda memoria solicitada en base a variable global y variable local

### DISCUSSION THREAD

**Discussion by maromero-21** [IIC2133-PUC/2024-2#6] [Preguntas Materia] [answered]
Hola! tengo una duda respecto a un código de las cápsulas de Valgrind que nos entregaron!
¿Por qué cuando el puntero C se define como variable global, Valgrind nos dice que la memoria se considera posiblemente perdida, mientras que si la defino como variable parte de la función main, la memoria se considera como definitivamente perdida?

Gracias por su tiempo!

Situación 1: puntero C como variable global
1. Código de esta situación: 
![image](https://github.com/user-attachments/assets/50dbb170-2284-4d54-82d0-63d4487cdbfc)
2. Output de Valgrind para este código:
![image](https://github.com/user-attachments/assets/add13f1c-308b-4489-b1d2-3d660d4cca8f)

Situación 2: puntero C como variable local de main
1. Código de esta situación:
![image](https://github.com/user-attachments/assets/237417ff-b0f3-4f96-808c-2a721b453808)
2. Output de Valgrind para este código:
![image](https://github.com/user-attachments/assets/734ab84f-febb-4a0f-9789-d67136a8056b)

**Comment by sebacarrascop** [comment, marked as answer]
La diferencia se debe a la visibilidad del puntero. 
Cuando C es global, Valgrind lo marca como "posiblemente perdida" porque aún podrías liberar la memoria en otro punto del programa. 
Cuando C es local a main, al finalizar la función, el puntero deja de existir, y la memoria se considera "definitivamente perdida" porque ya no hay forma de liberarla.

En resumen, mientras puedas acceder al puntero es posiblemente perdida, si ya no tienes acceso al puntero, es definitivamente perdida.
Saludos!

**Comment by maromero-21** [comment]
Muchas gracias por la respuesta! :)
